src/openforms/js/components/admin/form_design/logic/ActionSet.js

```diff
@@ -0,0 +1,177 @@
+import usePrevious from 'react-use/esm/usePrevious';
+import isEqual from 'lodash/isEqual';
+import PropTypes from "prop-types";
+import Trigger from "./Trigger";
+
+
+const emptyAction = {
+    actionType: '',
+    componentToChange: '',
+    componentValueSource: '',
+    componentProperty: '',
+    componentPropertyValue: '',
+    componentVariableValue: '',
+    componentLiteralValue: '',
+};
+
+const initialState = {
+    actions: []
+};
+
+const ACTION_SELECTION_ORDER = [
+    'actionType',
+    'componentToChange',
+    'componentValueSource',
+    'componentProperty',
+    'componentPropertyValue',
+    'componentVariableValue',
+    'componentLiteralValue',
+];
+
+
+const reducer = (draft, action) => {
+    switch(action.type) {
+        case 'ACTION_CHANGED': {
+            const {name, value, index} = action.payload;
+            draft.actions[index][name] = value;
+            console.log(index, name , value);
+
+            // clear the dependent fields if needed - e.g. if the component changes, all fields to the right change
+            const currentFieldIndex = ACTION_SELECTION_ORDER.indexOf(name);
+            const nextFieldNames = ACTION_SELECTION_ORDER.slice(currentFieldIndex + 1);
+            for (const name of nextFieldNames) {
+                draft.actions[index][name] = emptyAction[name];
+            }
+            break;
+        }
+        case 'ACTION_ADDED': {
+            draft.actions.push(emptyAction);
+            break;
+        }
+        case 'ACTION_DELETED': {
+            const {index} = action.payload;
+            const updatedActions = [...draft.actions];
+            updatedActions.splice(index, 1);
+            draft.actions = updatedActions;
+            break;
+        }
+        default: {
+            throw new Error(`Unknown action type: ${action.type}`);
+        }
+    }
+};
+
+const convertActionToJson = (action) => {
+        return {
+            component: action.componentToChange,
+            action: {
+                type: action.actionType,
+                property: {value: action.componentProperty},
+                // The data in 'value' needs to be valid jsonLogic
+                value: action.componentLiteralValue || {var: action.componentVariableValue},
+                state: action.componentPropertyValue,
+                // Selecting if the new component value should come from a literal or another component doesn't
+                // change anything in the JSON, only entering the literal or picking the component does.
+                // So, this 'source' attribute is here only to keep track of the value for the OperandTypeSelection
+                // dropdown, but is not used in the backend
+                source: action.componentValueSource
+        }
+    };
+};
+
+const parseJsonAction = (jsonAction) => {
+    let componentValueSource, componentLiteralValue, componentVariableValue;
+    if (jsonAction.action.value.var === undefined) {
```

Comment:
```suggestion
    if (jsonAction.action.value.var == null) {
```
this is a weird javascript thing where it will also match `undefined` (if you use the non-strict comparison), but a typical pattern. It catches both `undefined` and `null`.